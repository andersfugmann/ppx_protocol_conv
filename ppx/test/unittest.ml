module Driver = struct
  let name = "ppx_test"
  let serialize t = Marshal.to_string t []
  let deserialize t = Marshal.from_string t 0
  include Test_driver
  let of_driver_exn = of_test_exn
  let of_driver = of_test
  let to_driver = to_test
  let sexp_of_t : t -> Sexplib.Sexp.t = fun t ->
    Sexplib.Std.sexp_of_string (to_string_hum t)
end
module Unittest = Test.Unittest.Make (Driver)
let () = Unittest.run ()
