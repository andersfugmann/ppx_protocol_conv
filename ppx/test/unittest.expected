=== ppx_test.SingleElem ===
(List ())
=== ppx_test.SingleElem ===
(List ((Int 2)))
=== ppx_test.Longarray ===
(List ((Int 4) (Int 2) (Int 3) (Int 1)))
=== ppx_test.EmptyInsideRec ===
(Record ((c (String c)) (V (List ())) (a (String a))))
=== ppx_test.SingleInsideRec ===
(Record ((c (String c)) (V (List ((Int 2)))) (a (String a))))
=== ppx_test.MultiInsideRec ===
(Record
 ((c (String c)) (V (List ((Int 4) (Int 2) (Int 3) (Int 1)))) (a (String a))))
=== ppx_test.ArrayOfArrays ===
(Record ((a (List ((List ((Int 2) (Int 3))) (List ((Int 4) (Int 5))))))))
=== ppx_test.ArrayOfArrays2 ===
(List
 ((List ()) (List ((List ()) (List ((Int 2))) (List ((Int 3) (Int 4)))))
  (List ((List ()))) (List ((List ((Int 2)))))))
=== ppx_test.Tuple ===
(List
 ((List
   ((Int 10) (List ((Int 20) (Int 30) (Int 40)))
    (List ((String s50) (String s60) (String s70)))
    (List
     ((List ((Int 100) (Int 200))) (List ((Int 300) (Int 400)))
      (List ((Int 500) (Int 600)))))))
  (List
   ((Int 11) (List ((Int 21) (Int 31) (Int 41)))
    (List ((String s51) (String s61) (String s71)))
    (List
     ((List ((Int 101) (Int 201))) (List ((Int 301) (Int 401)))
      (List ((Int 501) (Int 601)))))))
  (List
   ((Int 12) (List ((Int 22) (Int 32) (Int 42)))
    (List ((String s52) (String s62) (String s72)))
    (List
     ((List ((Int 102) (Int 202))) (List ((Int 302) (Int 402)))
      (List ((Int 502) (Int 602)))))))
  (List
   ((Int 13) (List ((Int 23) (Int 33) (Int 43)))
    (List ((String s53) (String s63) (String s73)))
    (List
     ((List ((Int 103) (Int 203))) (List ((Int 303) (Int 403)))
      (List ((Int 503) (Int 603)))))))))
=== ppx_test.Any ===
(Record
 ((u (Record ((ua (Int 7)))))
  (v
   (Variant B
    ((List ((Int 5) (Int 6) (Int 7))) (List ((Int 10) (Int 11) (Int 12))))))
  (z (Int 101)) (record (Record ((y (String string)) (x (Int 5)))))
  (varray
   (List
    ((Variant B
      ((List ((Int 5) (Int 6) (Int 7))) (List ((Int 10) (Int 11) (Int 12)))))
     (Variant B
      ((List ((Int 5) (Int 6) (Int 7))) (List ((Int 10) (Int 11) (Int 12)))))
     (Variant B
      ((List ((Int 5) (Int 6) (Int 7))) (List ((Int 10) (Int 11) (Int 12))))))))
  (vlist
   (List
    ((Variant B
      ((List ((Int 5) (Int 6) (Int 7))) (List ((Int 10) (Int 11) (Int 12)))))
     (Variant B
      ((List ((Int 5) (Int 6) (Int 7))) (List ((Int 10) (Int 11) (Int 12)))))
     (Variant B
      ((List ((Int 5) (Int 6) (Int 7))) (List ((Int 10) (Int 11) (Int 12))))))))
  (tuple (List ((Int 5) (String protocol) (Bool false)))) (intref (Int 4))
  (intoption (Option ((Int 100)))) (intlist (List ((Int 3) (Int 4) (Int 5))))
  (bytes (Bytes bytes)) (string2 (String 1)) (string (String string))
  (float (Float 3.14)) (nativeint (Nativeint 20)) (int64 (Int64 10))
  (int32 (Int32 5)) (int (Int 2)) (char (Char x)) (bool (Bool true))))
=== ppx_test.Record ===
(Record
 ((t_il (List ((Int 1000) (Int 1001) (Int 1002))))
  (t_tl
   (List
    ((List ((Int 100) (Int 101) (List ((String s100) (String s101)))))
     (List ((Int 110) (Int 111) (List ((String s110) (String s111)))))
     (List ((Int 120) (Int 121) (List ((String s120) (String s121)))))
     (List ((Int 130) (Int 131) (List ((String s130) (String s131)))))
     (List ((Int 140) (Int 141) (List ((String s140) (String s141))))))))
  (t_t (List ((Int 100) (Int 101) (List ((String s100) (String s101))))))
  (t_i (Int 1000)) (t_bl (List ()))
  (t_b
   (Record
    ((b_al
      (List
       ((Record ((a_string (String s7)) (a_int (Int 7))))
        (Record ((a_string (String s8)) (a_int (Int 8))))
        (Record ((a_string (String s9)) (a_int (Int 9)))))))
     (b_a (Record ((a_string (String s6)) (a_int (Int 6)))))
     (b_string (String s5)) (b_int (Int 5)))))
  (t_al
   (List
    ((Record ((a_string (String s2)) (a_int (Int 2))))
     (Record ((a_string (String s3)) (a_int (Int 3))))
     (Record ((a_string (String s4)) (a_int (Int 4)))))))
  (t_a (Record ((a_string (String s1)) (a_int (Int 1)))))))
=== ppx_test.list ===
(Record ((a (List ((Int 1) (Int 2) (Int 3))))))
=== ppx_test.Lists ===
(Record
 ((l
   (List
    ((Variant A ((List ((Int 1) (Int 2) (Int 3)))))
     (Variant B
      ((List
        ((List ((Int 1) (Int 2))) (List ((Int 3) (Int 4) (Int 5)))
         (List ((Int 2)))))
       (List ((Int 3) (Int 1))) (Int 5)))
     (Variant C
      ((List ((Int 1) (Int 2) (Int 3))) (List ((Int 3) (Int 4) (Int 5))))))))
  (c (List ((Int 100) (Int 101) (Int 102) (Int 103))))
  (b
   (List
    ((List ((Int 8) (Int 9))) (List ((Int 10) (Int 20) (Int 30) (Int 40))))))
  (a
   (List
    ((List ((Int 1) (Int 2) (Int 3))) (List ())
     (List ((Int 10) (Int 20) (Int 30) (Int 40)))
     (List ((Int 100) (Int 101))))))))
=== ppx_test.array ===
(Record ((a (List ((Int 1) (Int 2) (Int 3))))))
=== ppx_test.EmptyList ===
(List ())
=== ppx_test.SingleElem ===
(List ((Int 2)))
=== ppx_test.Longlist ===
(List ((Int 4) (Int 3) (Int 2) (Int 1)))
=== ppx_test.EmptyInsideRec ===
(Record ((c (String c)) (V (List ())) (a (String a))))
=== ppx_test.SingleInsideRec ===
(Record ((c (String c)) (V (List ((Int 2)))) (a (String a))))
=== ppx_test.MultiInsideRec ===
(Record
 ((c (String c)) (V (List ((Int 4) (Int 2) (Int 3) (Int 1)))) (a (String a))))
=== ppx_test.ListOfLists ===
(Record ((a (List ((List ((Int 2) (Int 3))) (List ((Int 4) (Int 5))))))))
=== ppx_test.ListOfLists2 ===
(List
 ((List ()) (List ((List ()) (List ((Int 2))) (List ((Int 3) (Int 4)))))
  (List ((List ()))) (List ((List ((Int 2)))))))
=== ppx_test.Nonrec ===
(Variant A
 ((Variant Cons ((Int 4) (Variant Cons ((Int 3) (Variant Nil ())))))))
=== ppx_test.Nonrec2 ===
(Variant Cons ((Int 4) (Variant Cons ((Int 3) (Variant Nil ())))))
=== ppx_test.None ===
(Option ())
=== ppx_test.Some None ===
(Option ((Option ())))
=== ppx_test.Some Some None ===
(Option ((Option ((Option ())))))
=== ppx_test.Some Some Some Unit ===
(Option ((Option ((Option ((Option (Unit))))))))
=== ppx_test.simple ===
(Int 5)
=== ppx_test.record ===
(Record ((a (Int 5))))
=== ppx_test.multiple ===
(List ((Int 5) (String 5) (Bool true)))
=== ppx_test.reference ===
(Record ((a (Record ((a (Int 5)))))))
=== ppx_test.recursive ===
(Record ((c (Record ((b (Record ((a (Int 5))))))))))
=== ppx_test.Simple ===
(List
 ((Variant A ()) (Variant B ((Int 5))) (Variant C ((List ((Int 6) (Int 7)))))
  (Variant D ((List ((Int 8) (Int 9)))))))
=== ppx_test.Tree ===
(Variant Node
 ((List
   ((Variant Node ((List ((Variant Leaf ()) (Int 3) (Variant Leaf ())))))
    (Int 10) (Variant Leaf ())))))
=== ppx_test.MutualRecursion ===
(Variant T1
 ((Variant V
   ((Variant T
     ((Variant V
       ((Variant V1 ((Variant V1 ((Variant V1 ((Variant V0 ((Int 5)))))))))))))))))
=== ppx_test.InsideRec ===
(Record ((c (String c)) (V (Variant A ())) (a (String a))))
=== ppx_test.RecordList ===
(Record
 ((objects (List ((Record ((key (Int 1)))) (Record ((key (Int 2)))))))))
=== ppx_test.SimpleRecord ===
(Record
 ((HostId (String SDsd)) (RequestId (String sdfsd)) (Endpoint (Option ()))
  (Bucket (Option ())) (Message (String Message)) (Code (String Error))))
=== ppx_test.Test_sig ===
(Record
 ((x
   (Variant A
    ((List
      ((Variant A ((Int 7))) (Int 7) (Int 7) (Variant B ((Float 0.7)))
       (Int 7))))))))
=== ppx_test.Test_sig2 ===
(List
 ((Int 1) (Float 2) (String 3.0) Unit (Variant A ((Int 1)))
  (Record ((c (String 3.0)) (b (Float 2)) (a (Int 1))))
  (Variant A ((Int 1)))))
=== ppx_test.S3 ===
(Record
 ((Contents
   (List
    ((Record ((ETag (String Etag)) (StorageClass (Variant STANDARD ())))))))
  (Prefix (Option ((String prefix))))))
=== ppx_test.Types ===
(Record
 ((baz
   (Record
    ((y_yd (Variant Variant_two1 ((Int 1))))
     (y_yc
      (Option
       ((List ((String three) (List ((Int 100) (Int 200) (Int 300))))))))
     (y_b (List ((String two) (List ((Int 10) (Int 20) (Int 30))))))
     (y_a (Int 2)))))
  (bar (String true)) (foo (Int 1))))
=== ppx_test.Some Some Some true ===
(Option ((Option ((Option ((Bool true)))))))
=== ppx_test.Some Some None ===
(Option ((Option ((Option ())))))
=== ppx_test.Some None ===
(Option ((Option ())))
=== ppx_test.None ===
(Option ())
=== ppx_test.Some Some Some true ===
(Record ((a (Option ((Option ((Option ((Bool true))))))))))
=== ppx_test.Some Some None ===
(Record ((a (Option ((Option ((Option ()))))))))
=== ppx_test.Some None ===
(Record ((a (Option ((Option ()))))))
=== ppx_test.None ===
(Record ((a (Option ()))))
=== ppx_test.unit option option list option option ===
(Option
 ((Option
   ((List ((Option ((Option (Unit)))) (Option ((Option ()))) (Option ())))))))
=== ppx_test.confuse deserialization by using reserved word ===
(Record ((o (Record ((option (Option ((Option ((Bool true)))))))))))
=== ppx_test.Simple ===
(List
 ((Variant A ()) (Variant B ((Int 5))) (Variant C ((Int 6) (Int 7)))
  (Variant D ((List ((Int 8) (Int 9)))))))
=== ppx_test.Tuple ===
(Variant A ((List ((Int 3) (Int 4)))))
=== ppx_test.Tree ===
(Variant Node
 ((Variant Node ((Variant Leaf ()) (Int 3) (Variant Leaf ()))) (Int 10)
  (Variant Leaf ())))
=== ppx_test.MutualRecursion ===
(Variant T1
 ((Variant V
   ((Variant T
     ((Variant V
       ((Variant V1 ((Variant V1 ((Variant V1 ((Variant V0 ((Int 5)))))))))))))))))
=== ppx_test.InsideRec ===
(Record ((c (String c)) (V (Variant A ())) (a (String a))))
=== ppx_test.InlineRecord ===
(Variant A ((Record ((a (String a))))))
=== ppx_test.InlineRecord2 ===
(Variant aa
 ((Record
   ((b (Variant aa ((Record ((b (Variant B ((Int 5)))) (A (String a)))))))
    (A (String a))))))
=== ppx_test.Poly ===
(Variant aaa ((Int 5)))
=== ppx_test.Option.Ok ===
(Variant Ok ((Int 2)))
=== ppx_test.Option.Error ===
(Variant Error ((String "Error string")))
